### PR TITLE
fix(cancelOrder): Move cancel handler to Order component

### PR DESCRIPTION
**Pull request by flickz**
Resolves https://github.com/opencollective/opencollective/issues/2732

### CODE REVIEW

**Comment by flickz** [comment]
But this doesn't exactly fix, it's more of code maintenance that happens to fix the issue.

File: components/expenses/Order.js [hunk 1]
```diff
@@ -26,7 +30,7 @@ class Order extends React.Component {
     includeHostedCollectives: PropTypes.bool,
     LoggedInUser: PropTypes.object,
     intl: PropTypes.object.isRequired,
-    onClickCancel: PropTypes.func,
+    markPendingOrderAsExpired: PropTypes.func,
```
Comment (Betree) [inline components/expenses/Order.js]: If prop is mandatory then let's make it explicit

```suggestion
    markPendingOrderAsExpired: PropTypes.func.isRequired,
```

File: components/expenses/Order.js [hunk 2]
```diff
@@ -265,15 +296,61 @@ class Order extends React.Component {
...
+            {this.state.error}
+          </Container>
+        )}
       </div>
     );
   }
 }
 
-export default injectIntl(Order);
+const markPendingOrderAsExpiredQuery = gql`
```
Comment (Betree) [inline components/expenses/Order.js]:
```suggestion
const markPendingOrderAsExpiredMutation = gql`
```

File: components/expenses/Order.js [hunk 3]
```diff
@@ -265,15 +296,61 @@ class Order extends React.Component {
           {order.status === 'PENDING' && canMarkOrderAsPaid && (
             <Flex>
               <MarkOrderAsPaidBtn order={order} collective={order.collective} />
-              <StyledButton bg="red.500" color="#fff" onClick={() => this.props.onClickCancel(order.id)}>
+              <StyledButton
+                bg="red.500"
+                color="#fff"
```
Comment (Betree) [inline components/expenses/Order.js]: Let's rather use `buttonStyle="danger"`, this doesn't look very good

image

File: components/expenses/Order.js [hunk 4]
```diff
@@ -45,6 +51,16 @@ class Order extends React.Component {
       active: { id: 'order.active', defaultMessage: 'active' },
       cancelled: { id: 'order.cancelled', defaultMessage: 'cancelled' },
       expired: { id: 'order.expired', defaultMessage: 'expired' },
+      'cancelOrder.modal.header': {
+        id: 'cancelOrder.modal.header',
+        defaultMessage: 'Cancel Order',
+      },
+      'cancelOrder.modal.body': {
+        id: 'cancelOrder.modal.body',
+        defaultMessage: 'Are you sure you want to cancel this expense?',
```
Comment (Betree) [inline components/expenses/Order.js]:
```suggestion
        defaultMessage: 'Are you sure you want to cancel this order?',
```

File: components/expenses/Order.js [hunk 5]
```diff
@@ -45,6 +51,16 @@ class Order extends React.Component {
...
       expired: { id: 'order.expired', defaultMessage: 'expired' },
+      'cancelOrder.modal.header': {
+        id: 'cancelOrder.modal.header',
+        defaultMessage: 'Cancel Order',
+      },
+      'cancelOrder.modal.body': {
+        id: 'cancelOrder.modal.body',
+        defaultMessage: 'Are you sure you want to cancel this order?',
+      },
+      no: { id: 'no', defaultMessage: 'No' },
```
Comment (Betree) [inline components/expenses/Order.js]: Not required anymore

```suggestion
```

File: components/expenses/Order.js [hunk 6]
```diff
@@ -45,6 +51,16 @@ class Order extends React.Component {
...
+      yes: { id: 'yes', defaultMessage: 'Yes' },
```
Comment (Betree) [inline components/expenses/Order.js]:
```suggestion
```

File: components/expenses/Order.js [hunk 7]
```diff
@@ -264,15 +295,58 @@ class Order extends React.Component {
           {order.status === 'PENDING' && canMarkOrderAsPaid && (
             <Flex>
               <MarkOrderAsPaidBtn order={order} collective={order.collective} />
-              <StyledButton bg="red.500" color="#fff" onClick={() => this.props.onClickCancel(order.id)}>
+              <StyledButton
+                bg="red.500"
```
Comment (Betree) [inline components/expenses/Order.js]: Not required with buttonStyle=danger

```suggestion
```

**Pull request merged**